### PR TITLE
Feat/tet 849/card

### PR DESCRIPTION
**Pull request by mwleklinskiVL**
## Changes:
- Create Card component
- Add story with an example card content
- Write tests

### CODE REVIEW

**Comment by adrian-potepa** [comment]
There should be border: 

Zrzut ekranu 2024-02-27 o 11 43 43

**Review by adrian-potepa — CHANGES_REQUESTED** [review]
Small refactor is needed; dot notation components as props configuration (`header`, `footer`, `children`).

File: src/components/Card/CardHeader/CardHeader.props.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import type { AvatarProps } from '@/components/Avatar';
+import type { BareButtonProps } from '@/components/Button/Button.props';
+
+export type CardHeaderProps = {
+  title: string;
+  description?: string;
+  beforeComponent?: BeforeComponentProps;
+  styles: CardHeaderConfig;
+  actions?: BareButtonProps[];
```
Comment (adrian-potepa) [inline src/components/Card/CardHeader/CardHeader.props.ts:13]: We shouldn't allow to pass all possible props here, there should be limitations with size.

File: src/components/Card/CardHeader/CardHeader.props.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+};
+type BeforeComponentProps =
+  | {
+      icon: IconProps;
+    }
+  | {
+      avatar: AvatarProps;
```
Comment (adrian-potepa) [inline src/components/Card/CardHeader/CardHeader.props.ts:20]: Same story, it should be always rounded shape, medium size and low emphasis.

File: src/components/Card/CardFooter/CardFooter.props.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import type { CardFooterConfig } from '../Card.styles';
+
+import type { GhostButtonProps } from '@/components/Button/Button.props';
+
+export type CardFooterProps = {
+  styles: CardFooterConfig;
+  actions: GhostButtonProps[];
```
Comment (adrian-potepa) [inline src/components/Card/CardFooter/CardFooter.props.ts:7]: We need some limitations here.

**Review by adrian-potepa — CHANGES_REQUESTED** [review]
Required add some config props limitations.

File: src/components/Card/CardHeader/CardHeader.props.ts (hunk 1 above)
Comment (mwleklinskiVL) [reply src/components/Card/CardHeader/CardHeader.props.ts:13]: okej

Comment (mwleklinskiVL) [reply src/components/Card/CardHeader/CardHeader.props.ts:14]: temporarily on hold

File: src/components/Card/CardFooter/CardFooter.props.ts (hunk 3 above)
Comment (mwleklinskiVL) [reply src/components/Card/CardFooter/CardFooter.props.ts:8]: on hold for time being

**Review by adrian-potepa — APPROVED** [review]

**Pull request merged**